:seedling: Issue template for preparing new minor release

adilGhaffarDev opened the pull request:
<!-- please add an icon to the title of this PR (see https://sigs.k8s.io/cluster-api/CONTRIBUTING.md#contributing-a-patch), and delete this line and similar ones -->
<!-- the icon will be either ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🌱 (:seedling:, minor or other) -->

**What this PR does / why we need it**:
This PR is adding an issue template for preparing a new minor release. It will make it easier to create tracking issue.

kashifest commented:
/approve

tuminoid commented on .github/ISSUE_TEMPLATE/prepare_minor_release.md:
```diff
@@ -0,0 +1,25 @@
+
+* [ ] Add/edit JJBs to acommodate new build jobs and PR jobs for new release branches. [Prior art](https://gerrit.nordix.org/c/infra/cicd/+/20875)
+* [ ] Add Prow jobs for new release branches in Project-infra repo [Prior art](https://github.com/metal3-io/project-infra/pull/697)
+* [ ] Prepare dev-env to acommodate release-1.7 related changes and configurations [Prior art](https://github.com/metal3-io/metal3-dev-env/pull/1381)
```
I think we should not put past release versions in the template.

adilGhaffarDev replied:
removed versions, can you check again

tuminoid reviewed (commented):
/hold

tuminoid commented on .github/ISSUE_TEMPLATE/prepare_minor_release.md:
```diff
@@ -0,0 +1,25 @@
+* [ ] Add Prow jobs for new release branches in Project-infra repo [Prior art](https://github.com/metal3-io/project-infra/pull/697)
+* [ ] Prepare dev-env to acommodate new release related changes and configurations [Prior art](https://github.com/metal3-io/metal3-dev-env/pull/1381)
+* [ ] Update Metal3 book. [Prior art](https://github.com/metal3-io/metal3-docs/pull/413)
+* [ ] Release Ironic, also check new image is created in quay. [Prior art](https://github.com/metal3-io/ironic-image/tags)
```
Ironic-image release process is not part of CAPM3 minor release process. It is released separately when upstream Ironic has bugfix branch available. Maybe mark it as optional ?

adilGhaffarDev replied:
fixed.

tuminoid commented on .github/ISSUE_TEMPLATE/prepare_minor_release.md:
```diff
@@ -0,0 +1,25 @@
+* [ ] Prepare dev-env to acommodate new release related changes and configurations [Prior art](https://github.com/metal3-io/metal3-dev-env/pull/1381)
+* [ ] Update Metal3 book. [Prior art](https://github.com/metal3-io/metal3-docs/pull/413)
+* [ ] Release Ironic, also check new image is created in quay. [Prior art](https://github.com/metal3-io/ironic-image/tags)
+* [ ] Pin Ironic in BMO and release BMO, also check new image is created in quay. [Prior art](https://github.com/metal3-io/baremetal-operator/pull/1679)
```
IPAM mentions branching out etc, how is BMO any different?

adilGhaffarDev replied:
fixed

tuminoid commented on .github/ISSUE_TEMPLATE/prepare_minor_release.md:
```diff
@@ -0,0 +1,25 @@
+* [ ] Update Metal3 book. [Prior art](https://github.com/metal3-io/metal3-docs/pull/413)
+* [ ] Release Ironic, also check new image is created in quay. [Prior art](https://github.com/metal3-io/ironic-image/tags)
+* [ ] Pin Ironic in BMO and release BMO, also check new image is created in quay. [Prior art](https://github.com/metal3-io/baremetal-operator/pull/1679)
+* [ ] Release IPAM (Branch out, add branch protection and required tests, also check new image is created in quay).
```
Why were documenting IPAM/BMO/Ironic releasing in CAPM3 repo, shouldn't we have this tracking issue in more generic repo, like docs or dev-env or community?  Most of these steps do not related to CAPM3 after all.

adilGhaffarDev replied:
I am not sure what is the best place for this. If docs seems like a better place I can add it there. I added it here because the CAPM3 minor release is most of the time dependent on these releases.

tuminoid replied:
CAPM template should be about things done in CAPM3, similar to releasing.md is repo specific. Then in metal3-docs there is one that covers the common/extra steps. IMO these issue templates should be the same.

tuminoid commented on .github/ISSUE_TEMPLATE/prepare_minor_release.md:
```diff
@@ -0,0 +1,25 @@
+* [ ] Bump IPAM and BMO in CAPM3[Prior art](https://github.com/metal3-io/cluster-api-provider-metal3/pull/1605)
+* [ ] Release CAPM3 (Branch out, add branch protection and required tests, also check new image is created in quay).
+* [ ] Check CI if new releases are tested properly or not
+* [ ] Update CAPM3,IPAM and BMO README with the new e2e triggers. [Prior art][CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/pull/1618), [IPAM](https://github.com/metal3-io/ip-address-manager/pull/504),[BMO](https://github.com/metal3-io/baremetal-operator/pull/1694)
```
```suggestion
* [ ] Update CAPM3, IPAM and BMO README.md with the new e2e triggers. [Prior art][CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/pull/1618), [IPAM](https://github.com/metal3-io/ip-address-manager/pull/504),[BMO](https://github.com/metal3-io/baremetal-operator/pull/1694)
```

adilGhaffarDev replied:
fixed

tuminoid reviewed (commented):
NItting, but also questioning should this be somewhere else than CAPM3?

Sunnatillo commented:
@adilGhaffarDev Let's make this template get merged. This would be great to make release process smoother.

adilGhaffarDev commented:
> @adilGhaffarDev Let's make this template get merged. This would be great to make the release process smoother.

addressed the comments and also rearranged some tasks. Let me know if anything is missed.

mquhuy commented on .github/ISSUE_TEMPLATE/prepare_minor_release.md:
```diff
@@ -0,0 +1,35 @@
+* [ ] Uplift CAPI to latest minor in CAPM3, also check migration guide: [Migration guide for providers](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/providers/migrations). [Prior art CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/pull/1604).
+IMPORTANT: Always read migration guide and make sure to do the changes accordingly in CAPM3 and IPAM.
+* [x] Uplift k8s to 1.31 in CAPM3 and metal3-dev-env repo. [Prior art metal3-dev-env](https://github.com/metal3-io/metal3-dev-env/pull/1389), [Prior art CAPM3](https://github.com/metal3-io/cluster-api-provider-metal3/pull/1622)
+* [x] Uplift k8s to 1.31 in image building pipeline. [Prior art jjb](https://gerrit.nordix.org/c/infra/cicd/+/22205) [Prior art pipeline](https://github.com/metal3-io/project-infra/pull/836)
```
These 2 uplifts still refer to specific k8s versions, also remove the ticks.

mquhuy reviewed (commented):
Nit

tuminoid commented:
This is marked for 1.9, but it is just template for creating a release. We'll have new automation from IPAM coming in soon, so I think we should move it to 1.10. @adilGhaffarDev  is that ok?

adilGhaffarDev commented:
> This is marked for 1.9, but it is just template for creating a release. We'll have new automation from IPAM coming in soon, so I think we should move it to 1.10. @adilGhaffarDev is that ok?

yes we can move it to 1.10
